{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "compound-egypt",
   "metadata": {},
   "source": [
    "# Running lots of BLASTs\n",
    "Run BLASTs on 8 ENTREZ IDs and store in hits data frame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "fatal-footage",
   "metadata": {},
   "outputs": [],
   "source": [
    "import math\n",
    "import os\n",
    "import requests\n",
    "\n",
    "from Bio.Blast import NCBIWWW\n",
    "from Bio.Blast import NCBIXML\n",
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "accredited-korea",
   "metadata": {},
   "source": [
    "### Create pandas data frame for results, manually add ENTREZID's"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "conceptual-sarah",
   "metadata": {},
   "outputs": [],
   "source": [
    "data1 = pd.read_csv('data/001.csv')\n",
    "#Make sure ENTREZID column is integers not floats, or else NCBI cannot read it\n",
    "eids001 = [939978, 939970, 939444, 939454, 939466, 939456, 940002, 939969, 938032, 940000, 938032]\n",
    "\n",
    "selected_columns = data1[['GENENAME', 'ENTREZID', 'Log2fc_(GC)v(FLT)', 'P.value_(GC)v(FLT)', 'Adj.p.value_(GC)v(FLT)']]\n",
    "bioproject_df = pd.DataFrame(selected_columns)[0:10]\n",
    "protein_df = pd.DataFrame(selected_columns)[0:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "falling-approach",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENENAME</th>\n",
       "      <th>ENTREZID</th>\n",
       "      <th>Log2fc_(GC)v(FLT)</th>\n",
       "      <th>P.value_(GC)v(FLT)</th>\n",
       "      <th>Adj.p.value_(GC)v(FLT)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chromosomal replication initiator informationa...</td>\n",
       "      <td>939978.0</td>\n",
       "      <td>0.320344</td>\n",
       "      <td>9.332727e-05</td>\n",
       "      <td>4.157477e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DNA polymerase III (beta subunit)</td>\n",
       "      <td>939970.0</td>\n",
       "      <td>0.578609</td>\n",
       "      <td>5.564729e-10</td>\n",
       "      <td>7.000177e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RNA binding protein involved in ribosome matur...</td>\n",
       "      <td>939444.0</td>\n",
       "      <td>0.295755</td>\n",
       "      <td>1.011347e-04</td>\n",
       "      <td>4.478243e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RecA filament-DNA complex stabilisation, ssDNA...</td>\n",
       "      <td>939454.0</td>\n",
       "      <td>-0.047216</td>\n",
       "      <td>4.950332e-01</td>\n",
       "      <td>6.112680e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>regulator of extracellular matrix formation</td>\n",
       "      <td>939466.0</td>\n",
       "      <td>0.043596</td>\n",
       "      <td>5.978884e-01</td>\n",
       "      <td>6.994584e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>DNA gyrase (subunit B)</td>\n",
       "      <td>939456.0</td>\n",
       "      <td>0.079820</td>\n",
       "      <td>2.375809e-01</td>\n",
       "      <td>3.488090e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>DNA gyrase (subunit A)</td>\n",
       "      <td>940002.0</td>\n",
       "      <td>0.042476</td>\n",
       "      <td>6.033868e-01</td>\n",
       "      <td>7.036599e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>939969.0</td>\n",
       "      <td>0.572560</td>\n",
       "      <td>3.194062e-03</td>\n",
       "      <td>9.416315e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>inosine-monophosphate dehydrogenase</td>\n",
       "      <td>938032.0</td>\n",
       "      <td>0.092622</td>\n",
       "      <td>5.086308e-01</td>\n",
       "      <td>6.225034e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>D-alanyl-D-alanine carboxypeptidase (penicilli...</td>\n",
       "      <td>940000.0</td>\n",
       "      <td>0.398396</td>\n",
       "      <td>3.413016e-04</td>\n",
       "      <td>1.303955e-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            GENENAME  ENTREZID  \\\n",
       "0  chromosomal replication initiator informationa...  939978.0   \n",
       "1                  DNA polymerase III (beta subunit)  939970.0   \n",
       "2  RNA binding protein involved in ribosome matur...  939444.0   \n",
       "3  RecA filament-DNA complex stabilisation, ssDNA...  939454.0   \n",
       "4        regulator of extracellular matrix formation  939466.0   \n",
       "5                             DNA gyrase (subunit B)  939456.0   \n",
       "6                             DNA gyrase (subunit A)  940002.0   \n",
       "7                               hypothetical protein  939969.0   \n",
       "8                inosine-monophosphate dehydrogenase  938032.0   \n",
       "9  D-alanyl-D-alanine carboxypeptidase (penicilli...  940000.0   \n",
       "\n",
       "   Log2fc_(GC)v(FLT)  P.value_(GC)v(FLT)  Adj.p.value_(GC)v(FLT)  \n",
       "0           0.320344        9.332727e-05            4.157477e-04  \n",
       "1           0.578609        5.564729e-10            7.000177e-09  \n",
       "2           0.295755        1.011347e-04            4.478243e-04  \n",
       "3          -0.047216        4.950332e-01            6.112680e-01  \n",
       "4           0.043596        5.978884e-01            6.994584e-01  \n",
       "5           0.079820        2.375809e-01            3.488090e-01  \n",
       "6           0.042476        6.033868e-01            7.036599e-01  \n",
       "7           0.572560        3.194062e-03            9.416315e-03  \n",
       "8           0.092622        5.086308e-01            6.225034e-01  \n",
       "9           0.398396        3.413016e-04            1.303955e-03  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bioproject_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "potential-exposure",
   "metadata": {},
   "outputs": [],
   "source": [
    "# --Doesn't work\n",
    "def create_hits_df(db_hits_eidindex):\n",
    "    \"\"\"\n",
    "    Creates a dataframe for the NCBI BLAST hits results\n",
    "    \"\"\"\n",
    "    db_hits_eidindex = pd.DataFrame()\n",
    "    db_hits_eidindex['hits_EID'] = ''\n",
    "    db_hits_eidindex['hits_e_value'] = ''\n",
    "    db_hits_eidindex['hits_score'] = ''\n",
    "    db_hits_eidindex['normalized_score_%'] = ''\n",
    "    return db_hits_eidindex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "vietnamese-token",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hits_EID</th>\n",
       "      <th>hits_e_value</th>\n",
       "      <th>hits_score</th>\n",
       "      <th>normalized_score_%</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [hits_EID, hits_e_value, hits_score, normalized_score_%]\n",
       "Index: []"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Initiate results dataframes -- Doesn't work\n",
    "\n",
    "create_hits_df('protein_hits_0')\n",
    "create_hits_df('protein_hits_1')\n",
    "create_hits_df('protein_hits_2')\n",
    "create_hits_df('protein_hits_3')\n",
    "create_hits_df('protein_hits_4')\n",
    "create_hits_df('protein_hits_5')\n",
    "create_hits_df('protein_hits_6')\n",
    "create_hits_df('protein_hits_7')\n",
    "\n",
    "create_hits_df('pdb_hits_0')\n",
    "create_hits_df('pdb_hits_1')\n",
    "create_hits_df('pdb_hits_2')\n",
    "create_hits_df('pdb_hits_3')\n",
    "create_hits_df('pdb_hits_4')\n",
    "create_hits_df('pdb_hits_5')\n",
    "create_hits_df('pdb_hits_6')\n",
    "create_hits_df('pdb_hits_7')\n",
    "\n",
    "create_hits_df('bioproject_hits_0')\n",
    "create_hits_df('bioproject_hits_1')\n",
    "create_hits_df('bioproject_hits_2')\n",
    "create_hits_df('bioproject_hits_3')\n",
    "create_hits_df('bioproject_hits_4')\n",
    "create_hits_df('bioproject_hits_5')\n",
    "create_hits_df('bioproject_hits_6')\n",
    "create_hits_df('bioproject_hits_7')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "elder-pontiac",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create hits databases manually filled with zeros\n",
    "\n",
    "column_names = ['hits_EID', 'hits_e_value', 'hits_score', 'normalized_score_%']\n",
    "\n",
    "protein_hits_0 = pd.DataFrame(np.zeros((50, 4)), columns=column_names)\n",
    "protein_hits_1 = pd.DataFrame(np.zeros((50, 4)), columns=column_names)\n",
    "protein_hits_2 = pd.DataFrame(np.zeros((50, 4)), columns=column_names)\n",
    "protein_hits_3 = pd.DataFrame(np.zeros((50, 4)), columns=column_names)\n",
    "protein_hits_4 = pd.DataFrame(np.zeros((50, 4)), columns=column_names)\n",
    "protein_hits_5 = pd.DataFrame(np.zeros((50, 4)), columns=column_names)\n",
    "protein_hits_6 = pd.DataFrame(np.zeros((50, 4)), columns=column_names)\n",
    "protein_hits_7 = pd.DataFrame(np.zeros((50, 4)), columns=column_names)\n",
    "\n",
    "pdb_hits_0 = pd.DataFrame(np.zeros((50, 4)), columns=column_names)\n",
    "pdb_hits_1 = pd.DataFrame(columns=column_names)\n",
    "pdb_hits_2 = pd.DataFrame(columns=column_names)\n",
    "pdb_hits_3 = pd.DataFrame(columns=column_names)\n",
    "pdb_hits_4 = pd.DataFrame(columns=column_names)\n",
    "pdb_hits_5 = pd.DataFrame(columns=column_names)\n",
    "pdb_hits_6 = pd.DataFrame(columns=column_names)\n",
    "pdb_hits_7 = pd.DataFrame(columns=column_names)\n",
    "\n",
    "bioproject_hits_0 = pd.DataFrame(columns=column_names)\n",
    "bioproject_hits_1 = pd.DataFrame(columns=column_names)\n",
    "bioproject_hits_2 = pd.DataFrame(columns=column_names)\n",
    "bioproject_hits_3 = pd.DataFrame(columns=column_names)\n",
    "bioproject_hits_4 = pd.DataFrame(columns=column_names)\n",
    "bioproject_hits_5 = pd.DataFrame(columns=column_names)\n",
    "bioproject_hits_6 = pd.DataFrame(columns=column_names)\n",
    "bioproject_hits_7 = pd.DataFrame(columns=column_names)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "imperial-wyoming",
   "metadata": {},
   "source": [
    "### Function to BLAST for EID's and save in data frame per NCBI DB as .XML and convert pd.DF to .csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "hazardous-senator",
   "metadata": {},
   "outputs": [],
   "source": [
    "def blast_eid_for_dataframe(ENTREZID, save_file_xml, eid_index, ncbi_database, results_dataframe, csv):\n",
    "    \"\"\"\n",
    "    Function: Take EID, BLAST in a given NCBI database\n",
    "    ENTREZID: ENTREZID from 001.csv as added to bioproject_df, float\n",
    "    save_file_xml: file to save to as ncbidb_index.xml, string\n",
    "    eid_index: index of the ENTREZID, float\n",
    "    ncbi_database: the NCBI database to be used for BLASTing, string\n",
    "    results_dataframe: ncbi_hits, as created previously, string\n",
    "    csv: filename as ncbidb_index.csv to store hits into, string\n",
    "    \"\"\"\n",
    "    eid = ENTREZID\n",
    "    result_handle = NCBIWWW.qblast('blastp', ncbi_database, eid)\n",
    "    with open(save_file_xml, 'w+') as save_to:\n",
    "        save_to.write(result_handle.read())\n",
    "        result_handle.close()    \n",
    "    result_handle = open(save_file_xml)\n",
    "    blast_record = NCBIXML.read(result_handle)\n",
    "\n",
    "    E_VALUE_THRESH = 0.0000000000000000000000000000000000000000000000004\n",
    "    for alignment in blast_record.alignments:\n",
    "        for hsp in alignment.hsps:\n",
    "            if hsp.expect < E_VALUE_THRESH:\n",
    "                results_dataframe['hits_EID'][eid_index] = alignment.title\n",
    "                results_dataframe['hits_e_value'][eid_index] = hsp.expect\n",
    "                results_dataframe['hits_score'][eid_index] = hsp.score\n",
    "                results_dataframe['normalized_score_%'][eid_index] = hsp.score / results_dataframe['normalized_score_%'][1] * 100\n",
    "    \n",
    "    results_dataframe.to_csv(csv)\n",
    "    return "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "distinct-workstation",
   "metadata": {},
   "source": [
    "Test this function  \n",
    "Results: it seems to have been running"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "efficient-breast",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pdb|1HDG|O The Crystal Structure Of Holo-glyceraldehyde-3-phosphate Dehydrogenase From The Hyperthermophilic Bacterium Thermotoga Maritima At 2.5 Angstroms Resolution [Thermotoga maritima] >pdb|1HDG|Q The Crystal Structure Of Holo-glyceraldehyde-3-phosphate Dehydrogenase From The Hyperthermophilic Bacterium Thermotoga Maritima At 2.5 Angstroms Resolution [Thermotoga maritima]\n",
      "0.0\n",
      "pdb|1GD1|O Structure Of Holo-Glyceraldehyde-3-Phosphate Dehydrogenase From Bacillus Stearothermophilus At 1.8 Angstroms Resolution [Geobacillus stearothermophilus] >pdb|1GD1|P Structure Of Holo-Glyceraldehyde-3-Phosphate Dehydrogenase From Bacillus Stearothermophilus At 1.8 Angstroms Resolution [Geobacillus stearothermophilus] >pdb|1GD1|Q Structure Of Holo-Glyceraldehyde-3-Phosphate Dehydrogenase From Bacillus Stearothermophilus At 1.8 Angstroms Resolution [Geobacillus stearothermophilus] >pdb|1GD1|R Structure Of Holo-Glyceraldehyde-3-Phosphate Dehydrogenase From Bacillus Stearothermophilus At 1.8 Angstroms Resolution [Geobacillus stearothermophilus] >pdb|2GD1|O Coenzyme-Induced Conformational Changes In Glyceraldehyde-3- Phosphate Dehydrogenase From Bacillus Stearothermophillus [Geobacillus stearothermophilus] >pdb|2GD1|P Coenzyme-Induced Conformational Changes In Glyceraldehyde-3- Phosphate Dehydrogenase From Bacillus Stearothermophillus [Geobacillus stearothermophilus] >pdb|2GD1|Q Coenzyme-Induced Conformational Changes In Glyceraldehyde-3- Phosphate Dehydrogenase From Bacillus Stearothermophillus [Geobacillus stearothermophilus] >pdb|2GD1|R Coenzyme-Induced Conformational Changes In Glyceraldehyde-3- Phosphate Dehydrogenase From Bacillus Stearothermophillus [Geobacillus stearothermophilus] >pdb|3CMC|O Thioacylenzyme intermediate of Bacillus stearothermophilus phosphorylating GAPDH [Geobacillus stearothermophilus] >pdb|3CMC|P Thioacylenzyme intermediate of Bacillus stearothermophilus phosphorylating GAPDH [Geobacillus stearothermophilus] >pdb|3CMC|Q Thioacylenzyme intermediate of Bacillus stearothermophilus phosphorylating GAPDH [Geobacillus stearothermophilus] >pdb|3CMC|R Thioacylenzyme intermediate of Bacillus stearothermophilus phosphorylating GAPDH [Geobacillus stearothermophilus]\n",
      "2.09943e-155\n",
      "pdb|1NQ5|A Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|1NQ5|C Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|1NQ5|O Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|1NQ5|Q Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|1NQO|A Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ser Complexed With Nad+ and D-Glyceraldehyde-3-Phosphate [Geobacillus stearothermophilus] >pdb|1NQO|C Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ser Complexed With Nad+ and D-Glyceraldehyde-3-Phosphate [Geobacillus stearothermophilus] >pdb|1NQO|O Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ser Complexed With Nad+ and D-Glyceraldehyde-3-Phosphate [Geobacillus stearothermophilus] >pdb|1NQO|Q Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ser Complexed With Nad+ and D-Glyceraldehyde-3-Phosphate [Geobacillus stearothermophilus]\n",
      "4.44302e-154\n",
      "pdb|1NPT|O Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 replaced by Ala complexed with NAD+ [Geobacillus stearothermophilus] >pdb|1NPT|P Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 replaced by Ala complexed with NAD+ [Geobacillus stearothermophilus] >pdb|1NPT|Q Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 replaced by Ala complexed with NAD+ [Geobacillus stearothermophilus] >pdb|1NPT|R Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 replaced by Ala complexed with NAD+ [Geobacillus stearothermophilus] >pdb|1NQA|O Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ala Complexed With Nad+ and D-Glyceraldehyde-3-Phosphate [Geobacillus stearothermophilus] >pdb|1NQA|P Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ala Complexed With Nad+ and D-Glyceraldehyde-3-Phosphate [Geobacillus stearothermophilus] >pdb|1NQA|Q Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ala Complexed With Nad+ and D-Glyceraldehyde-3-Phosphate [Geobacillus stearothermophilus] >pdb|1NQA|R Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Cys 149 Replaced By Ala Complexed With Nad+ and D-Glyceraldehyde-3-Phosphate [Geobacillus stearothermophilus]\n",
      "5.65196e-154\n",
      "pdb|1DBV|O Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Asp 32 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|1DBV|P Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Asp 32 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|1DBV|Q Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Asp 32 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|1DBV|R Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Asp 32 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|2DBV|O Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Asp 32 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nadp+ [Geobacillus stearothermophilus] >pdb|2DBV|P Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Asp 32 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nadp+ [Geobacillus stearothermophilus] >pdb|2DBV|Q Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Asp 32 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nadp+ [Geobacillus stearothermophilus] >pdb|2DBV|R Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Asp 32 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nadp+ [Geobacillus stearothermophilus]\n",
      "2.90045e-152\n",
      "pdb|3DBV|O Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Leu 33 Replaced By Thr, Thr 34 Replaced By Gly, Asp 36 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|3DBV|P Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Leu 33 Replaced By Thr, Thr 34 Replaced By Gly, Asp 36 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|3DBV|Q Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Leu 33 Replaced By Thr, Thr 34 Replaced By Gly, Asp 36 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|3DBV|R Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Leu 33 Replaced By Thr, Thr 34 Replaced By Gly, Asp 36 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nad+ [Geobacillus stearothermophilus] >pdb|4DBV|O Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Leu 33 Replaced By Thr, Thr 34 Replaced By Gly, Asp 36 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nadp+ [Geobacillus stearothermophilus] >pdb|4DBV|P Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Leu 33 Replaced By Thr, Thr 34 Replaced By Gly, Asp 36 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nadp+ [Geobacillus stearothermophilus] >pdb|4DBV|Q Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Leu 33 Replaced By Thr, Thr 34 Replaced By Gly, Asp 36 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nadp+ [Geobacillus stearothermophilus] >pdb|4DBV|R Glyceraldehyde-3-Phosphate Dehydrogenase Mutant With Leu 33 Replaced By Thr, Thr 34 Replaced By Gly, Asp 36 Replaced By Gly, Leu 187 Replaced By Ala, And Pro 188 Replaced By Ser Complexed With Nadp+ [Geobacillus stearothermophilus]\n",
      "2.32975e-150\n",
      "pdb|2EP7|A Structural study of Project ID aq_1065 from Aquifex aeolicus VF5 [Aquifex aeolicus] >pdb|2EP7|B Structural study of Project ID aq_1065 from Aquifex aeolicus VF5 [Aquifex aeolicus]\n",
      "2.92327e-145\n",
      "pdb|1VC2|A Crystal structure of Glyceraldehyde 3-Phosphate Dehydrogenase from Thermus thermophilus HB8 [Thermus thermophilus]\n",
      "3.55986e-142\n",
      "pdb|1CER|A Determinants Of Enzyme Thermostability Observed In The Molecular Structure Of Thermus Aquaticus D-Glyceraldehyde- 3-Phosphate Dehydrogenase At 2.5 Angstroms Resolution [Thermus aquaticus] >pdb|1CER|B Determinants Of Enzyme Thermostability Observed In The Molecular Structure Of Thermus Aquaticus D-Glyceraldehyde- 3-Phosphate Dehydrogenase At 2.5 Angstroms Resolution [Thermus aquaticus] >pdb|1CER|C Determinants Of Enzyme Thermostability Observed In The Molecular Structure Of Thermus Aquaticus D-Glyceraldehyde- 3-Phosphate Dehydrogenase At 2.5 Angstroms Resolution [Thermus aquaticus] >pdb|1CER|D Determinants Of Enzyme Thermostability Observed In The Molecular Structure Of Thermus Aquaticus D-Glyceraldehyde- 3-Phosphate Dehydrogenase At 2.5 Angstroms Resolution [Thermus aquaticus] >pdb|1CER|O Determinants Of Enzyme Thermostability Observed In The Molecular Structure Of Thermus Aquaticus D-Glyceraldehyde- 3-Phosphate Dehydrogenase At 2.5 Angstroms Resolution [Thermus aquaticus] >pdb|1CER|P Determinants Of Enzyme Thermostability Observed In The Molecular Structure Of Thermus Aquaticus D-Glyceraldehyde- 3-Phosphate Dehydrogenase At 2.5 Angstroms Resolution [Thermus aquaticus] >pdb|1CER|Q Determinants Of Enzyme Thermostability Observed In The Molecular Structure Of Thermus Aquaticus D-Glyceraldehyde- 3-Phosphate Dehydrogenase At 2.5 Angstroms Resolution [Thermus aquaticus] >pdb|1CER|R Determinants Of Enzyme Thermostability Observed In The Molecular Structure Of Thermus Aquaticus D-Glyceraldehyde- 3-Phosphate Dehydrogenase At 2.5 Angstroms Resolution [Thermus aquaticus]\n",
      "3.35997e-140\n",
      "pdb|2G82|A High Resolution Structures of Thermus aquaticus Glyceraldehyde-3-Phosphate Dehydrogenase: Role of 220's Loop Motion in Catalysis [Thermus aquaticus] >pdb|2G82|B High Resolution Structures of Thermus aquaticus Glyceraldehyde-3-Phosphate Dehydrogenase: Role of 220's Loop Motion in Catalysis [Thermus aquaticus] >pdb|2G82|C High Resolution Structures of Thermus aquaticus Glyceraldehyde-3-Phosphate Dehydrogenase: Role of 220's Loop Motion in Catalysis [Thermus aquaticus] >pdb|2G82|D High Resolution Structures of Thermus aquaticus Glyceraldehyde-3-Phosphate Dehydrogenase: Role of 220's Loop Motion in Catalysis [Thermus aquaticus] >pdb|2G82|O High Resolution Structures of Thermus aquaticus Glyceraldehyde-3-Phosphate Dehydrogenase: Role of 220's Loop Motion in Catalysis [Thermus aquaticus] >pdb|2G82|P High Resolution Structures of Thermus aquaticus Glyceraldehyde-3-Phosphate Dehydrogenase: Role of 220's Loop Motion in Catalysis [Thermus aquaticus] >pdb|2G82|Q High Resolution Structures of Thermus aquaticus Glyceraldehyde-3-Phosphate Dehydrogenase: Role of 220's Loop Motion in Catalysis [Thermus aquaticus] >pdb|2G82|R High Resolution Structures of Thermus aquaticus Glyceraldehyde-3-Phosphate Dehydrogenase: Role of 220's Loop Motion in Catalysis [Thermus aquaticus]\n",
      "1.95937e-138\n",
      "pdb|3GNQ|A Crystal structure of glyceraldehyde-3-phosphate dehydrogenase, type I from Burkholderia pseudomallei [Burkholderia pseudomallei 1710b] >pdb|3GNQ|B Crystal structure of glyceraldehyde-3-phosphate dehydrogenase, type I from Burkholderia pseudomallei [Burkholderia pseudomallei 1710b] >pdb|3GNQ|C Crystal structure of glyceraldehyde-3-phosphate dehydrogenase, type I from Burkholderia pseudomallei [Burkholderia pseudomallei 1710b] >pdb|3GNQ|D Crystal structure of glyceraldehyde-3-phosphate dehydrogenase, type I from Burkholderia pseudomallei [Burkholderia pseudomallei 1710b] >pdb|3GNQ|E Crystal structure of glyceraldehyde-3-phosphate dehydrogenase, type I from Burkholderia pseudomallei [Burkholderia pseudomallei 1710b] >pdb|3GNQ|F Crystal structure of glyceraldehyde-3-phosphate dehydrogenase, type I from Burkholderia pseudomallei [Burkholderia pseudomallei 1710b] >pdb|3GNQ|G Crystal structure of glyceraldehyde-3-phosphate dehydrogenase, type I from Burkholderia pseudomallei [Burkholderia pseudomallei 1710b] >pdb|3GNQ|H Crystal structure of glyceraldehyde-3-phosphate dehydrogenase, type I from Burkholderia pseudomallei [Burkholderia pseudomallei 1710b]\n",
      "4.92074e-133\n",
      "pdb|3DOC|A Crystal Structure of TrkA glyceraldehyde-3-phosphate dehydrogenase from Brucella melitensis [Brucella abortus 2308] >pdb|3DOC|B Crystal Structure of TrkA glyceraldehyde-3-phosphate dehydrogenase from Brucella melitensis [Brucella abortus 2308] >pdb|3DOC|C Crystal Structure of TrkA glyceraldehyde-3-phosphate dehydrogenase from Brucella melitensis [Brucella abortus 2308] >pdb|3DOC|D Crystal Structure of TrkA glyceraldehyde-3-phosphate dehydrogenase from Brucella melitensis [Brucella abortus 2308]\n",
      "7.59162e-133\n",
      "pdb|3HJA|A Crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Borrelia burgdorferi [Borreliella burgdorferi B31] >pdb|3HJA|B Crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Borrelia burgdorferi [Borreliella burgdorferi B31] >pdb|3HJA|C Crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Borrelia burgdorferi [Borreliella burgdorferi B31] >pdb|3HJA|D Crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Borrelia burgdorferi [Borreliella burgdorferi B31]\n",
      "1.5945e-131\n",
      "pdb|3L0D|A Crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Bartonella henselae with bound NAD [Bartonella henselae] >pdb|3L0D|B Crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Bartonella henselae with bound NAD [Bartonella henselae]\n",
      "4.26256e-130\n",
      "pdb|3B1J|A Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with CP12 in the presence of copper from Synechococcus elongatus [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|3B1J|B Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with CP12 in the presence of copper from Synechococcus elongatus [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|3B1K|A Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with CP12 in the absence of copper from Synechococcus elongatus [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|3B1K|B Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with CP12 in the absence of copper from Synechococcus elongatus [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|3B1K|G Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with CP12 in the absence of copper from Synechococcus elongatus [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|3B1K|H Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with CP12 in the absence of copper from Synechococcus elongatus [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|3B20|A Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with NADfrom Synechococcus elongatus' [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|3B20|B Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with NADfrom Synechococcus elongatus' [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|3B20|O Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with NADfrom Synechococcus elongatus' [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|3B20|P Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with NADfrom Synechococcus elongatus' [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|3B20|Q Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with NADfrom Synechococcus elongatus' [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|3B20|R Crystal structure of Glyceraldehyde-3-Phosphate Dehydrogenase complexed with NADfrom Synechococcus elongatus' [Synechococcus elongatus PCC 7942 = FACHB-805]\n",
      "1.80244e-128\n",
      "pdb|4DIB|A The crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Bacillus anthracis str. Sterne [Bacillus anthracis str. Sterne] >pdb|4DIB|B The crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Bacillus anthracis str. Sterne [Bacillus anthracis str. Sterne] >pdb|4DIB|C The crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Bacillus anthracis str. Sterne [Bacillus anthracis str. Sterne] >pdb|4DIB|D The crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Bacillus anthracis str. Sterne [Bacillus anthracis str. Sterne] >pdb|4DIB|E The crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Bacillus anthracis str. Sterne [Bacillus anthracis str. Sterne] >pdb|4DIB|F The crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Bacillus anthracis str. Sterne [Bacillus anthracis str. Sterne] >pdb|4DIB|G The crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Bacillus anthracis str. Sterne [Bacillus anthracis str. Sterne] >pdb|4DIB|H The crystal structure of glyceraldehyde-3-phosphate dehydrogenase from Bacillus anthracis str. Sterne [Bacillus anthracis str. Sterne]\n",
      "2.28668e-127\n",
      "pdb|2D2I|A Crystal Structure of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. complexed with Nadp+ [Synechococcus sp.] >pdb|2D2I|B Crystal Structure of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. complexed with Nadp+ [Synechococcus sp.] >pdb|2D2I|O Crystal Structure of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. complexed with Nadp+ [Synechococcus sp.] >pdb|2D2I|P Crystal Structure of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. complexed with Nadp+ [Synechococcus sp.] >pdb|2D2I|Q Crystal Structure of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. complexed with Nadp+ [Synechococcus sp.] >pdb|2D2I|R Crystal Structure of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. complexed with Nadp+ [Synechococcus sp.] >pdb|2DUU|A Crystal Structure of apo-form of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|2DUU|B Crystal Structure of apo-form of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|2DUU|O Crystal Structure of apo-form of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|2DUU|P Crystal Structure of apo-form of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|2DUU|Q Crystal Structure of apo-form of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. [Synechococcus elongatus PCC 7942 = FACHB-805] >pdb|2DUU|R Crystal Structure of apo-form of NADP-Dependent Glyceraldehyde-3-Phosphate Dehydrogenase from Synechococcus Sp. [Synechococcus elongatus PCC 7942 = FACHB-805]\n",
      "1.10068e-126\n",
      "pdb|3K73|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3K73|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3K73|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3K73|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3L6O|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3L6O|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3L6O|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3L6O|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LC2|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LC2|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LC2|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LC2|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252]\n",
      "2.33928e-125\n",
      "pdb|3LVF|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LVF|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LVF|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LVF|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252]\n",
      "2.88559e-125\n",
      "pdb|3VAZ|A Chain A, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3VAZ|B Chain B, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3VAZ|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3VAZ|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3VAZ|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3VAZ|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252]\n",
      "3.53699e-125\n",
      "pdb|3LC7|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LC7|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LC7|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LC7|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252]\n",
      "3.59393e-125\n",
      "pdb|6FZI|A Crystal Structure of a Clostridial Dehydrogenase at 2.55 Angstroems Resolution [Clostridium perfringens SM101] >pdb|6FZI|B Crystal Structure of a Clostridial Dehydrogenase at 2.55 Angstroems Resolution [Clostridium perfringens SM101] >pdb|6FZI|C Crystal Structure of a Clostridial Dehydrogenase at 2.55 Angstroems Resolution [Clostridium perfringens SM101] >pdb|6FZI|D Crystal Structure of a Clostridial Dehydrogenase at 2.55 Angstroems Resolution [Clostridium perfringens SM101]\n",
      "4.15055e-125\n",
      "pdb|3LC1|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LC1|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LC1|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3LC1|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252]\n",
      "3.5823e-124\n",
      "pdb|5VMT|A Crystal structure of a glyceraldehyde-3-phosphate dehydrogenase from Neisseria gonorrhoeae bound to NAD [Neisseria gonorrhoeae NCCP11945] >pdb|5VMT|B Crystal structure of a glyceraldehyde-3-phosphate dehydrogenase from Neisseria gonorrhoeae bound to NAD [Neisseria gonorrhoeae NCCP11945] >pdb|5VMT|C Crystal structure of a glyceraldehyde-3-phosphate dehydrogenase from Neisseria gonorrhoeae bound to NAD [Neisseria gonorrhoeae NCCP11945] >pdb|5VMT|D Crystal structure of a glyceraldehyde-3-phosphate dehydrogenase from Neisseria gonorrhoeae bound to NAD [Neisseria gonorrhoeae NCCP11945] >pdb|5VMT|E Crystal structure of a glyceraldehyde-3-phosphate dehydrogenase from Neisseria gonorrhoeae bound to NAD [Neisseria gonorrhoeae NCCP11945] >pdb|5VMT|F Crystal structure of a glyceraldehyde-3-phosphate dehydrogenase from Neisseria gonorrhoeae bound to NAD [Neisseria gonorrhoeae NCCP11945] >pdb|5VMT|G Crystal structure of a glyceraldehyde-3-phosphate dehydrogenase from Neisseria gonorrhoeae bound to NAD [Neisseria gonorrhoeae NCCP11945] >pdb|5VMT|H Crystal structure of a glyceraldehyde-3-phosphate dehydrogenase from Neisseria gonorrhoeae bound to NAD [Neisseria gonorrhoeae NCCP11945]\n",
      "7.09602e-124\n",
      "pdb|3HQ4|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3HQ4|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3HQ4|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3HQ4|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3KV3|O Chain O, Gapdh [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3KV3|P Chain P, Gapdh [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3KV3|Q Chain Q, Gapdh [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3KV3|R Chain R, Gapdh [Staphylococcus aureus subsp. aureus MRSA252]\n",
      "7.52451e-124\n",
      "pdb|5T73|A Chain A, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|5T73|B Chain B, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|5T73|C Chain C, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|5T73|D Chain D, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252]\n",
      "1.19001e-123\n",
      "pdb|3K9Q|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3K9Q|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3K9Q|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3K9Q|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3L4S|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3L4S|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3L4S|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3L4S|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252]\n",
      "1.49653e-123\n",
      "pdb|6GFO|A cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GFO|B cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GFO|C cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GFO|D cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GFP|A cyanobacterial GAPDH with NADP bound [Thermosynechococcus elongatus BP-1] >pdb|6GFP|B cyanobacterial GAPDH with NADP bound [Thermosynechococcus elongatus BP-1] >pdb|6GFQ|A cyanobacterial GAPDH with NAD and CP12 bound [Thermosynechococcus elongatus BP-1] >pdb|6GFQ|B cyanobacterial GAPDH with NAD and CP12 bound [Thermosynechococcus elongatus BP-1] >pdb|6GFR|A cyanobacterial GAPDH with NAD [Thermosynechococcus elongatus BP-1] >pdb|6GFR|B cyanobacterial GAPDH with NAD [Thermosynechococcus elongatus BP-1] >pdb|6GG7|A cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GG7|B cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GHL|A cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GHL|B cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GHL|C cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GHL|D cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GVE|A GAPDH-CP12-PRK complex [Thermosynechococcus elongatus BP-1] >pdb|6GVE|D GAPDH-CP12-PRK complex [Thermosynechococcus elongatus BP-1] >pdb|6GVE|E GAPDH-CP12-PRK complex [Thermosynechococcus elongatus BP-1] >pdb|6GVE|F GAPDH-CP12-PRK complex [Thermosynechococcus elongatus BP-1] >pdb|6GVE|I GAPDH-CP12-PRK complex [Thermosynechococcus elongatus BP-1] >pdb|6GVE|K GAPDH-CP12-PRK complex [Thermosynechococcus elongatus BP-1] >pdb|6GVE|N GAPDH-CP12-PRK complex [Thermosynechococcus elongatus BP-1] >pdb|6GVE|O GAPDH-CP12-PRK complex [Thermosynechococcus elongatus BP-1]\n",
      "6.00437e-123\n",
      "pdb|3KSD|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3KSD|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3KSD|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3KSD|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3KSZ|O Chain O, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3KSZ|P Chain P, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3KSZ|Q Chain Q, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252] >pdb|3KSZ|R Chain R, Glyceraldehyde-3-phosphate dehydrogenase 1 [Staphylococcus aureus subsp. aureus MRSA252]\n",
      "8.96129e-123\n",
      "pdb|3ZCX|A Structure of GAPDH from Thermosynechococcus elongatus [Thermosynechococcus elongatus] >pdb|3ZCX|B Structure of GAPDH from Thermosynechococcus elongatus [Thermosynechococcus elongatus] >pdb|3ZDF|A Structure of GAPDH with CP12 peptide from Thermosynechococcus elongatus [Thermosynechococcus elongatus] >pdb|3ZDF|B Structure of GAPDH with CP12 peptide from Thermosynechococcus elongatus [Thermosynechococcus elongatus] >pdb|4BOY|A Structure of GAPDH from Thermosynechococcus elongatus [Thermosynechococcus elongatus] >pdb|4BOY|B Structure of GAPDH from Thermosynechococcus elongatus [Thermosynechococcus elongatus] >pdb|6GHR|A cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GHR|B cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GHR|C cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1] >pdb|6GHR|D cyanobacterial GAPDH with full-length CP12 [Thermosynechococcus elongatus BP-1]\n",
      "9.32353e-123\n",
      "pdb|1RM3|A Crystal structure of mutant T33A of photosynthetic glyceraldehyde-3-phosphate dehydrogenase A4 isoform, complexed with NADP [Spinacia oleracea] >pdb|1RM3|B Crystal structure of mutant T33A of photosynthetic glyceraldehyde-3-phosphate dehydrogenase A4 isoform, complexed with NADP [Spinacia oleracea] >pdb|1RM3|O Crystal structure of mutant T33A of photosynthetic glyceraldehyde-3-phosphate dehydrogenase A4 isoform, complexed with NADP [Spinacia oleracea]\n",
      "1.03388e-122\n",
      "pdb|1NBO|A The dual coenzyme specificity of photosynthetic glyceraldehyde-3-phosphate dehydrogenase interpreted by the crystal structure of A4 isoform complexed with NAD [Spinacia oleracea] >pdb|1NBO|B The dual coenzyme specificity of photosynthetic glyceraldehyde-3-phosphate dehydrogenase interpreted by the crystal structure of A4 isoform complexed with NAD [Spinacia oleracea] >pdb|1NBO|O The dual coenzyme specificity of photosynthetic glyceraldehyde-3-phosphate dehydrogenase interpreted by the crystal structure of A4 isoform complexed with NAD [Spinacia oleracea] >pdb|1RM4|A Crystal structure of recombinant photosynthetic glyceraldehyde-3-phosphate dehydrogenase A4 isoform, complexed with NADP [Spinacia oleracea] >pdb|1RM4|B Crystal structure of recombinant photosynthetic glyceraldehyde-3-phosphate dehydrogenase A4 isoform, complexed with NADP [Spinacia oleracea] >pdb|1RM4|O Crystal structure of recombinant photosynthetic glyceraldehyde-3-phosphate dehydrogenase A4 isoform, complexed with NADP [Spinacia oleracea] >pdb|2HKI|A Crystal structure of photosynthetic glyceraldehyde-3-phosphate dehydrogenase A4 isoform [Spinacia oleracea] >pdb|2PKQ|P Crystal structure of the photosynthetic A2B2-glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKQ|R Crystal structure of the photosynthetic A2B2-glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKQ|S Crystal structure of the photosynthetic A2B2-glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea]\n",
      "1.06828e-122\n",
      "pdb|1RM5|A Crystal structure of mutant S188A of photosynthetic glyceraldehyde-3-phosphate dehydrogenase A4 isoform, complexed with NADP [Spinacia oleracea] >pdb|1RM5|B Crystal structure of mutant S188A of photosynthetic glyceraldehyde-3-phosphate dehydrogenase A4 isoform, complexed with NADP [Spinacia oleracea] >pdb|1RM5|O Crystal structure of mutant S188A of photosynthetic glyceraldehyde-3-phosphate dehydrogenase A4 isoform, complexed with NADP [Spinacia oleracea]\n",
      "1.20453e-122\n",
      "pdb|2PKR|A Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKR|B Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKR|C Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKR|D Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKR|H Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKR|I Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKR|L Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKR|M Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKR|O Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKR|P Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKR|Q Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea] >pdb|2PKR|R Crystal structure of (A+CTE)4 chimeric form of photosyntetic glyceraldehyde-3-phosphate dehydrogenase, complexed with NADP [Spinacia oleracea]\n",
      "6.00189e-122\n",
      "pdb|7JWK|A Crystal Structure of Glyceraldehyde-3-phosphate Dehydrogenase (GAPDH) from Mycoplasma genitalium with bound NAD [Mycoplasma genitalium G37] >pdb|7JWK|B Crystal Structure of Glyceraldehyde-3-phosphate Dehydrogenase (GAPDH) from Mycoplasma genitalium with bound NAD [Mycoplasma genitalium G37] >pdb|7JWK|C Crystal Structure of Glyceraldehyde-3-phosphate Dehydrogenase (GAPDH) from Mycoplasma genitalium with bound NAD [Mycoplasma genitalium G37] >pdb|7JWK|D Crystal Structure of Glyceraldehyde-3-phosphate Dehydrogenase (GAPDH) from Mycoplasma genitalium with bound NAD [Mycoplasma genitalium G37]\n",
      "1.99542e-121\n",
      "pdb|1JN0|A Crystal structure of the non-regulatory A4 isoform of spinach chloroplast glyceraldehyde-3-phosphate dehydrogenase complexed with NADP [Spinacia oleracea] >pdb|1JN0|B Crystal structure of the non-regulatory A4 isoform of spinach chloroplast glyceraldehyde-3-phosphate dehydrogenase complexed with NADP [Spinacia oleracea] >pdb|1JN0|O Crystal structure of the non-regulatory A4 isoform of spinach chloroplast glyceraldehyde-3-phosphate dehydrogenase complexed with NADP [Spinacia oleracea]\n",
      "8.02288e-121\n",
      "pdb|1OBF|O The crystal structure of Glyceraldehyde 3-phosphate Dehydrogenase from Alcaligenes xylosoxidans at 1.7A resolution. [Achromobacter xylosoxidans] >pdb|1OBF|P The crystal structure of Glyceraldehyde 3-phosphate Dehydrogenase from Alcaligenes xylosoxidans at 1.7A resolution. [Achromobacter xylosoxidans]\n",
      "2.23736e-120\n",
      "pdb|6NLX|A Crystal Structure of Glyceraldehyde-3-phosphate Dehydrogenase from Naegleria fowleri with bound NAD [Naegleria fowleri] >pdb|6NLX|B Crystal Structure of Glyceraldehyde-3-phosphate Dehydrogenase from Naegleria fowleri with bound NAD [Naegleria fowleri] >pdb|6NLX|C Crystal Structure of Glyceraldehyde-3-phosphate Dehydrogenase from Naegleria fowleri with bound NAD [Naegleria fowleri] >pdb|6NLX|D Crystal Structure of Glyceraldehyde-3-phosphate Dehydrogenase from Naegleria fowleri with bound NAD [Naegleria fowleri]\n",
      "2.01586e-119\n",
      "pdb|5M6D|A Streptococcus pneumoniae Glyceraldehyde-3-Phosphate Dehydrogenase (SpGAPDH) crystal structure [Streptococcus pneumoniae] >pdb|5M6D|B Streptococcus pneumoniae Glyceraldehyde-3-Phosphate Dehydrogenase (SpGAPDH) crystal structure [Streptococcus pneumoniae]\n",
      "5.20337e-119\n",
      "pdb|6FZH|A Crystal structure of a streptococcal dehydrogenase at 1.5 Angstroem resolution [Streptococcus pyogenes MGAS8232] >pdb|6FZH|B Crystal structure of a streptococcal dehydrogenase at 1.5 Angstroem resolution [Streptococcus pyogenes MGAS8232]\n",
      "2.90487e-118\n",
      "pdb|6ITE|O Crystal structure of group A Streptococcal surface dehydrogenase (SDH) [Streptococcus pyogenes] >pdb|6ITE|P Crystal structure of group A Streptococcal surface dehydrogenase (SDH) [Streptococcus pyogenes] >pdb|6ITE|Q Crystal structure of group A Streptococcal surface dehydrogenase (SDH) [Streptococcus pyogenes] >pdb|6ITE|R Crystal structure of group A Streptococcal surface dehydrogenase (SDH) [Streptococcus pyogenes]\n",
      "9.58397e-118\n",
      "pdb|5UTL|A Mutant Structures of Streptococcus Agalactiae GBS Glyceraldehyde-3-Phosphate Dehydrogenase (GAPDH) [Streptococcus agalactiae] >pdb|5UTL|B Mutant Structures of Streptococcus Agalactiae GBS Glyceraldehyde-3-Phosphate Dehydrogenase (GAPDH) [Streptococcus agalactiae] >pdb|5UTL|C Mutant Structures of Streptococcus Agalactiae GBS Glyceraldehyde-3-Phosphate Dehydrogenase (GAPDH) [Streptococcus agalactiae] >pdb|5UTL|D Mutant Structures of Streptococcus Agalactiae GBS Glyceraldehyde-3-Phosphate Dehydrogenase (GAPDH) [Streptococcus agalactiae]\n",
      "1.72328e-117\n",
      "pdb|5Y37|A Crystal structure of GBS GAPDH [Streptococcus agalactiae NEM316] >pdb|5Y37|B Crystal structure of GBS GAPDH [Streptococcus agalactiae NEM316] >pdb|5Y37|C Crystal structure of GBS GAPDH [Streptococcus agalactiae NEM316] >pdb|5Y37|D Crystal structure of GBS GAPDH [Streptococcus agalactiae NEM316]\n",
      "1.80659e-117\n",
      "pdb|3PYM|A Structure of GAPDH 3 from S.cerevisiae at 2.0 A resolution [Saccharomyces cerevisiae] >pdb|3PYM|B Structure of GAPDH 3 from S.cerevisiae at 2.0 A resolution [Saccharomyces cerevisiae]\n",
      "2.59245e-117\n",
      "pdb|6IEP|A GAPDH of Streptococcus agalactiae [Streptococcus agalactiae NEM316] >pdb|6IEP|B GAPDH of Streptococcus agalactiae [Streptococcus agalactiae NEM316] >pdb|6IEP|C GAPDH of Streptococcus agalactiae [Streptococcus agalactiae NEM316] >pdb|6IEP|D GAPDH of Streptococcus agalactiae [Streptococcus agalactiae NEM316]\n",
      "3.04905e-117\n",
      "pdb|4IQ8|A Crystal structure of glyceraldehyde-3-phosphate dehydrogenase 3 from Saccharomyces cerevisiae [Saccharomyces cerevisiae S288C]\n",
      "3.14129e-117\n",
      "pdb|5V8X|A Mutant Structures of Streptococcus Agalactiae GBS Glyceraldehyde-3-Phosphate Dehydrogenase (GAPDH) [Streptococcus agalactiae] >pdb|5V8X|B Mutant Structures of Streptococcus Agalactiae GBS Glyceraldehyde-3-Phosphate Dehydrogenase (GAPDH) [Streptococcus agalactiae] >pdb|5V8X|C Mutant Structures of Streptococcus Agalactiae GBS Glyceraldehyde-3-Phosphate Dehydrogenase (GAPDH) [Streptococcus agalactiae] >pdb|5V8X|D Mutant Structures of Streptococcus Agalactiae GBS Glyceraldehyde-3-Phosphate Dehydrogenase (GAPDH) [Streptococcus agalactiae]\n",
      "6.23687e-117\n",
      "pdb|5V8Y|A Mutant Structures of Streptococcus Agalactiae GBS Glyceraldehyde-3-Phosphate Dehydrogenase (GAPDH) [Streptococcus agalactiae] >pdb|5V8Y|B Mutant Structures of Streptococcus Agalactiae GBS Glyceraldehyde-3-Phosphate Dehydrogenase (GAPDH) [Streptococcus agalactiae]\n",
      "7.105e-117\n",
      "pdb|2I5P|O Crystal structure of glyceraldehyde-3-phosphate dehydrogenase isoform 1 from K. marxianus [Kluyveromyces marxianus] >pdb|2I5P|P Crystal structure of glyceraldehyde-3-phosphate dehydrogenase isoform 1 from K. marxianus [Kluyveromyces marxianus]\n",
      "7.20542e-117\n",
      "pdb|4QX6|A CRYSTAL STRUCTURE OF GLYCERALDEHYDE-3-PHOSPHATE DEHYDROGENASE FROM STREPTOCOCCUS AGALACTIAE NEM316 at 2.46 ANGSTROM RESOLUTION [Streptococcus agalactiae NEM316] >pdb|4QX6|B CRYSTAL STRUCTURE OF GLYCERALDEHYDE-3-PHOSPHATE DEHYDROGENASE FROM STREPTOCOCCUS AGALACTIAE NEM316 at 2.46 ANGSTROM RESOLUTION [Streptococcus agalactiae NEM316] >pdb|4QX6|C CRYSTAL STRUCTURE OF GLYCERALDEHYDE-3-PHOSPHATE DEHYDROGENASE FROM STREPTOCOCCUS AGALACTIAE NEM316 at 2.46 ANGSTROM RESOLUTION [Streptococcus agalactiae NEM316] >pdb|4QX6|D CRYSTAL STRUCTURE OF GLYCERALDEHYDE-3-PHOSPHATE DEHYDROGENASE FROM STREPTOCOCCUS AGALACTIAE NEM316 at 2.46 ANGSTROM RESOLUTION [Streptococcus agalactiae NEM316] >pdb|5JY6|A Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae] >pdb|5JY6|B Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae] >pdb|5JY6|C Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae] >pdb|5JY6|D Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae] >pdb|5JYE|A Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae] >pdb|5JYE|B Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae] >pdb|5JYE|C Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae] >pdb|5JYE|D Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae] >pdb|5JYF|A Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae] >pdb|5JYF|B Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae] >pdb|5JYF|C Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae] >pdb|5JYF|D Structures of Streptococcus agalactiae GBS GAPDH in different enzymatic states [Streptococcus agalactiae]\n",
      "7.42519e-117\n"
     ]
    }
   ],
   "source": [
    "result_handle = open('pdb_0.xml')\n",
    "blast_record = NCBIXML.read(result_handle)\n",
    "\n",
    "E_VALUE_THRESH = 0.0000000000000000000000000000000000000000000000004\n",
    "for alignment in blast_record.alignments:\n",
    "    for hsp in alignment.hsps:\n",
    "        if hsp.expect < E_VALUE_THRESH:\n",
    "            print(alignment.title)\n",
    "            print(hsp.expect)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "imported-relief",
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "int() argument must be a string, a bytes-like object or a number, not 'Alignment'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-74-51881c1a7135>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      6\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mhsp\u001b[0m \u001b[0;32min\u001b[0m \u001b[0malignment\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhsps\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mhsp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexpect\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0mE_VALUE_THRESH\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m             \u001b[0mpdb_hits_0\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0malignment\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'hits_EID'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0malignment\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtitle\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m             \u001b[0mpdb_hits_0\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0malignment\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'hits_EID'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhsp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexpect\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m             \u001b[0mpdb_hits_0\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0malignment\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'hits_EID'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhsp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscore\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    689\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    690\u001b[0m         \u001b[0miloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"iloc\"\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 691\u001b[0;31m         \u001b[0miloc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_setitem_with_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    692\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    693\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_validate_key\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_setitem_with_indexer\u001b[0;34m(self, indexer, value, name)\u001b[0m\n\u001b[1;32m   1610\u001b[0m                     \u001b[0;31m# so the object is the same\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1611\u001b[0m                     \u001b[0mindex\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1612\u001b[0;31m                     \u001b[0mlabels\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1613\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_mgr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreindex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlabels\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_mgr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1614\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_maybe_update_cacher\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclear\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/numeric.py\u001b[0m in \u001b[0;36minsert\u001b[0;34m(self, loc, item)\u001b[0m\n\u001b[1;32m    176\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobject\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 178\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    179\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    180\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_union\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msort\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36minsert\u001b[0;34m(self, loc, item)\u001b[0m\n\u001b[1;32m   5560\u001b[0m         \u001b[0;31m#  so self is never backed by an EA.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5561\u001b[0m         \u001b[0marr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 5562\u001b[0;31m         \u001b[0mitem\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_coerce_scalar_to_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   5563\u001b[0m         \u001b[0midx\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconcatenate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mitem\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5564\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0midx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36m_coerce_scalar_to_index\u001b[0;34m(self, item)\u001b[0m\n\u001b[1;32m   4189\u001b[0m             \u001b[0mdtype\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4190\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4191\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_attributes_dict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4192\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4193\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_validate_fill_value\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36m__new__\u001b[0;34m(cls, data, dtype, copy, name, tupleize_cols, **kwargs)\u001b[0m\n\u001b[1;32m    389\u001b[0m             \u001b[0;31m# other iterable of some kind\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    390\u001b[0m             \u001b[0msubarr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray_tuplesafe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mobject\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 391\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubarr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    392\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    393\u001b[0m     \"\"\"\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36m__new__\u001b[0;34m(cls, data, dtype, copy, name, tupleize_cols, **kwargs)\u001b[0m\n\u001b[1;32m    335\u001b[0m                 \u001b[0;31m# should not be coerced\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    336\u001b[0m                 \u001b[0;31m# GH 11836\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 337\u001b[0;31m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_maybe_cast_with_dtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    338\u001b[0m                 \u001b[0mdtype\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m  \u001b[0;31m# TODO: maybe not for object?\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    339\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36m_maybe_cast_with_dtype\u001b[0;34m(data, dtype, copy)\u001b[0m\n\u001b[1;32m   6008\u001b[0m             \u001b[0;32mpass\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6009\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 6010\u001b[0;31m             \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   6011\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mis_float_dtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6012\u001b[0m         \u001b[0minferred\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minfer_dtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mskipna\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: int() argument must be a string, a bytes-like object or a number, not 'Alignment'"
     ]
    }
   ],
   "source": [
    "result_handle = open('pdb_0.xml')\n",
    "blast_record = NCBIXML.read(result_handle)\n",
    "\n",
    "E_VALUE_THRESH = 0.0000000000000000000000000000000000000000000000004\n",
    "for alignment in blast_record.alignments:\n",
    "    for hsp in alignment.hsps:\n",
    "        if hsp.expect < E_VALUE_THRESH:\n",
    "            pdb_hits_0.loc[alignment,('hits_EID')] = alignment.title\n",
    "            pdb_hits_0.loc[alignment,('hits_EID')] = hsp.expect\n",
    "            pdb_hits_0.loc[alignment,('hits_EID')] = hsp.score\n",
    "            pdb_hits_0.loc[alignment,('hits_EID')] = hsp.score / pdb_hits_0['normalized_score_%'][0] * 100\n",
    "    \n",
    "pdb_hits_0.to_csv('pdb_1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "foreign-credits",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-71-d379b6bdde6b>:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  pdb_hits_0['hits_EID'][alignment] = alignment.title\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "int() argument must be a string, a bytes-like object or a number, not 'Alignment'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    942\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 943\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_with_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    944\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mKeyError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m_set_with_engine\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    975\u001b[0m         \u001b[0;31m# fails with AttributeError for IntervalIndex\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 976\u001b[0;31m         \u001b[0mloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    977\u001b[0m         \u001b[0mvalidate_numeric_casting\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.index.Int64Engine._check_type\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: <Bio.Blast.Record.Alignment object at 0x7f184ab37070>",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-71-d379b6bdde6b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      6\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mhsp\u001b[0m \u001b[0;32min\u001b[0m \u001b[0malignment\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhsps\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mhsp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexpect\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0mE_VALUE_THRESH\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m             \u001b[0mpdb_hits_0\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'hits_EID'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0malignment\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0malignment\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtitle\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m             \u001b[0mpdb_hits_0\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'hits_e_value'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0malignment\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhsp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexpect\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m             \u001b[0mpdb_hits_0\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'hits_score'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0malignment\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhsp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscore\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    949\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    950\u001b[0m                 \u001b[0;31m# GH#12862 adding a new key to the Series\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 951\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    952\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    953\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    689\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    690\u001b[0m         \u001b[0miloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"iloc\"\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 691\u001b[0;31m         \u001b[0miloc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_setitem_with_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    692\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    693\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_validate_key\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_setitem_with_indexer\u001b[0;34m(self, indexer, value, name)\u001b[0m\n\u001b[1;32m   1626\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1627\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mmissing\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1628\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_setitem_with_indexer_missing\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1629\u001b[0m                 \u001b[0;32mreturn\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1630\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_setitem_with_indexer_missing\u001b[0;34m(self, indexer, value)\u001b[0m\n\u001b[1;32m   1871\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndim\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1872\u001b[0m             \u001b[0mindex\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1873\u001b[0;31m             \u001b[0mnew_index\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1874\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1875\u001b[0m             \u001b[0;31m# we have a coerced indexer, e.g. a float\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/numeric.py\u001b[0m in \u001b[0;36minsert\u001b[0;34m(self, loc, item)\u001b[0m\n\u001b[1;32m    176\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobject\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 178\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    179\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    180\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_union\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msort\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36minsert\u001b[0;34m(self, loc, item)\u001b[0m\n\u001b[1;32m   5560\u001b[0m         \u001b[0;31m#  so self is never backed by an EA.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5561\u001b[0m         \u001b[0marr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 5562\u001b[0;31m         \u001b[0mitem\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_coerce_scalar_to_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   5563\u001b[0m         \u001b[0midx\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconcatenate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mitem\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5564\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0midx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36m_coerce_scalar_to_index\u001b[0;34m(self, item)\u001b[0m\n\u001b[1;32m   4189\u001b[0m             \u001b[0mdtype\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4190\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4191\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_attributes_dict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4192\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4193\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_validate_fill_value\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36m__new__\u001b[0;34m(cls, data, dtype, copy, name, tupleize_cols, **kwargs)\u001b[0m\n\u001b[1;32m    389\u001b[0m             \u001b[0;31m# other iterable of some kind\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    390\u001b[0m             \u001b[0msubarr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray_tuplesafe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mobject\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 391\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubarr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    392\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    393\u001b[0m     \"\"\"\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36m__new__\u001b[0;34m(cls, data, dtype, copy, name, tupleize_cols, **kwargs)\u001b[0m\n\u001b[1;32m    335\u001b[0m                 \u001b[0;31m# should not be coerced\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    336\u001b[0m                 \u001b[0;31m# GH 11836\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 337\u001b[0;31m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_maybe_cast_with_dtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    338\u001b[0m                 \u001b[0mdtype\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m  \u001b[0;31m# TODO: maybe not for object?\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    339\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36m_maybe_cast_with_dtype\u001b[0;34m(data, dtype, copy)\u001b[0m\n\u001b[1;32m   6008\u001b[0m             \u001b[0;32mpass\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6009\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 6010\u001b[0;31m             \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   6011\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mis_float_dtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6012\u001b[0m         \u001b[0minferred\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minfer_dtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mskipna\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: int() argument must be a string, a bytes-like object or a number, not 'Alignment'"
     ]
    }
   ],
   "source": [
    "result_handle = open('pdb_0.xml')\n",
    "blast_record = NCBIXML.read(result_handle)\n",
    "\n",
    "E_VALUE_THRESH = 0.0000000000000000000000000000000000000000000000004\n",
    "for alignment in blast_record.alignments:\n",
    "    for hsp in alignment.hsps:\n",
    "        if hsp.expect < E_VALUE_THRESH:\n",
    "            pdb_hits_0['hits_EID'][alignment] = alignment.title\n",
    "            pdb_hits_0['hits_e_value'][alignment] = hsp.expect\n",
    "            pdb_hits_0['hits_score'][alignment] = hsp.score\n",
    "            pdb_hits_0['normalized_score_%'][alignment] = hsp.score / pdb_hits_0['normalized_score_%'][0] * 100\n",
    "    \n",
    "pdb_hits_0.to_csv('pdb_1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "linear-connectivity",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "939970"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eids001[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "mexican-newport",
   "metadata": {},
   "outputs": [
    {
     "ename": "IndexError",
     "evalue": "index 1 is out of bounds for axis 0 with size 0",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    942\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 943\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_with_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    944\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mKeyError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m_set_with_engine\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    975\u001b[0m         \u001b[0;31m# fails with AttributeError for IntervalIndex\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 976\u001b[0;31m         \u001b[0mloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    977\u001b[0m         \u001b[0mvalidate_numeric_casting\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 1",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-59-de0efe9fcfcc>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mblast_eid_for_dataframe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0meids001\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'pdb_1.xml'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'pdb'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpdb_hits_1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'pdb_1.csv'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-39-9fbbd4e0b04c>\u001b[0m in \u001b[0;36mblast_eid_for_dataframe\u001b[0;34m(ENTREZID, save_file_xml, eid_index, ncbi_database, results_dataframe, csv)\u001b[0m\n\u001b[1;32m     21\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mhsp\u001b[0m \u001b[0;32min\u001b[0m \u001b[0malignment\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhsps\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     22\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mhsp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexpect\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0mE_VALUE_THRESH\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 23\u001b[0;31m                 \u001b[0mresults_dataframe\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'hits_EID'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0meid_index\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0malignment\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtitle\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     24\u001b[0m                 \u001b[0mresults_dataframe\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'hits_e_value'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0meid_index\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhsp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexpect\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m                 \u001b[0mresults_dataframe\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'hits_score'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0meid_index\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhsp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscore\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    946\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minferred_type\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"integer\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    947\u001b[0m                 \u001b[0;31m# positional setter\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 948\u001b[0;31m                 \u001b[0mvalues\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    949\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    950\u001b[0m                 \u001b[0;31m# GH#12862 adding a new key to the Series\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mIndexError\u001b[0m: index 1 is out of bounds for axis 0 with size 0"
     ]
    }
   ],
   "source": [
    "blast_eid_for_dataframe(eids001[1], 'pdb_1.xml', 1, 'pdb', pdb_hits_1, 'pdb_1.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "decreased-luxembourg",
   "metadata": {},
   "source": [
    "### BLAST the first 8 ENTREZIDs for 3 different databases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "activated-latin",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'str' object has no attribute 'to_csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-41-04c28acb6873>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mblast_eid_for_dataframe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0meids001\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_0.xml'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_hits_0'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_0.csv'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mblast_eid_for_dataframe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0meids001\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_1.xml'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_hits_1'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_1.csv'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mblast_eid_for_dataframe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0meids001\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_2.xml'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m2\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_hits_2'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_2.csv'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mblast_eid_for_dataframe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0meids001\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_3.xml'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m3\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_hits_3'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_3.csv'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mblast_eid_for_dataframe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0meids001\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_4.xml'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m4\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_hits_4'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein_4.csv'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-39-9fbbd4e0b04c>\u001b[0m in \u001b[0;36mblast_eid_for_dataframe\u001b[0;34m(ENTREZID, save_file_xml, eid_index, ncbi_database, results_dataframe, csv)\u001b[0m\n\u001b[1;32m     26\u001b[0m                 \u001b[0mresults_dataframe\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'normalized_score_%'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0meid_index\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhsp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscore\u001b[0m \u001b[0;34m/\u001b[0m \u001b[0mresults_dataframe\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'normalized_score_%'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m*\u001b[0m \u001b[0;36m100\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     27\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 28\u001b[0;31m     \u001b[0mresults_dataframe\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcsv\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     29\u001b[0m     \u001b[0;32mreturn\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'str' object has no attribute 'to_csv'"
     ]
    }
   ],
   "source": [
    "blast_eid_for_dataframe(eids001[0], 'protein_0.xml', 0, 'protein', 'protein_hits_0', 'protein_0.csv')\n",
    "blast_eid_for_dataframe(eids001[1], 'protein_1.xml', 1, 'protein', 'protein_hits_1', 'protein_1.csv')\n",
    "blast_eid_for_dataframe(eids001[2], 'protein_2.xml', 2, 'protein', 'protein_hits_2', 'protein_2.csv')\n",
    "blast_eid_for_dataframe(eids001[3], 'protein_3.xml', 3, 'protein', 'protein_hits_3', 'protein_3.csv')\n",
    "blast_eid_for_dataframe(eids001[4], 'protein_4.xml', 4, 'protein', 'protein_hits_4', 'protein_4.csv')\n",
    "blast_eid_for_dataframe(eids001[5], 'protein_5.xml', 5, 'protein', 'protein_hits_5', 'protein_5.csv')\n",
    "blast_eid_for_dataframe(eids001[6], 'protein_6.xml', 6, 'protein', 'protein_hits_6', 'protein_6.csv')\n",
    "blast_eid_for_dataframe(eids001[7], 'protein_7.xml', 7, 'protein', 'protein_hits_7', 'protein_7.csv')\n",
    "\n",
    "blast_eid_for_dataframe(eids001[0], 'pdb_0.xml', 0, 'pdb', 'pdb_hits_0', 'pdb_0.csv')\n",
    "blast_eid_for_dataframe(eids001[1], 'pdb_1.xml', 1, 'pdb', 'pdb_hits_1', 'pdb_1.csv')\n",
    "blast_eid_for_dataframe(eids001[2], 'pdb_2.xml', 2, 'pdb', 'pdb_hits_2', 'pdb_2.csv')\n",
    "blast_eid_for_dataframe(eids001[3], 'pdb_3.xml', 3, 'pdb', 'pdb_hits_3', 'pdb_3.csv')\n",
    "blast_eid_for_dataframe(eids001[4], 'pdb_4.xml', 4, 'pdb', 'pdb_hits_4', 'pdb_4.csv')\n",
    "blast_eid_for_dataframe(eids001[5], 'pdb_5.xml', 5, 'pdb', 'pdb_hits_5', 'pdb_5.csv')\n",
    "blast_eid_for_dataframe(eids001[6], 'pdb_6.xml', 6, 'pdb', 'pdb_hits_6', 'pdb_6.csv')\n",
    "blast_eid_for_dataframe(eids001[7], 'pdb_7.xml', 7, 'pdb', 'pdb_hits_7', 'pdb_7.csv')\n",
    "\n",
    "blast_eid_for_dataframe(eids001[0], 'bioproject_0.xml', 0, 'bioproject', 'bioproject_hits_0', 'bioproject_0.csv')\n",
    "blast_eid_for_dataframe(eids001[1], 'bioproject_1.xml', 1, 'bioproject', 'bioproject_hits_1', 'bioproject_1.csv')\n",
    "blast_eid_for_dataframe(eids001[2], 'bioproject_2.xml', 2, 'bioproject', 'bioproject_hits_2', 'bioproject_2.csv')\n",
    "blast_eid_for_dataframe(eids001[3], 'bioproject_3.xml', 3, 'bioproject', 'bioproject_hits_3', 'bioproject_3.csv')\n",
    "blast_eid_for_dataframe(eids001[4], 'bioproject_4.xml', 4, 'bioproject', 'bioproject_hits_4', 'bioproject_4.csv')\n",
    "blast_eid_for_dataframe(eids001[5], 'bioproject_5.xml', 5, 'bioproject', 'bioproject_hits_5', 'bioproject_5.csv')\n",
    "blast_eid_for_dataframe(eids001[6], 'bioproject_6.xml', 6, 'bioproject', 'bioproject_hits_6', 'bioproject_6.csv')\n",
    "blast_eid_for_dataframe(eids001[7], 'bioproject_7.xml', 7, 'bioproject', 'bioproject_hits_7', 'bioproject_7.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "broad-window",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
